src/features/views/components/ViewColumnDialog/ColumnGallery/index.tsx

```diff
@@ -67,14 +57,12 @@ const ColumnGallery: FunctionComponent<ColumnGalleryProps> = ({
       .search(searchString)
       .map((fuseResult) => fuseResult.item.key);
 
-    //get the choice objects with matching keys
-    const results = keys.map((key) =>
-      choices.find((choice) => choice.key === key)
-    );
-
-    setSearchResults(results);
+    //return the choice objects with matching keys
+    return keys.map((key) => choices.find((choice) => choice.key === key));
   };
 
+  const searchResults = useMemo(() => search(), [searchString]);
```

Comment:
The expression `() => search()` is synonymous with `search`, so this could be simplified further, but that's not something I think is necessary.